tensorboard/data/server/tf_record.rs

```diff
@@ -352,4 +383,46 @@ mod tests {
             other => panic!("{:?}", other),
         }
     }
+
+    #[test]
+    fn test_from_data() {
+        let test_cases = vec![
+            b"".to_vec(),
+            b"\x00".to_vec(),
+            b"the quick brown fox jumped over the lazy dog".to_vec(),
+        ];
+        for data in test_cases {
+            TfRecord::from_data(data).checksum().unwrap();
+        }
+    }
+
+    fn test_write_read_roundtrip(record: &TfRecord) {
+        let mut cursor = Cursor::new(Vec::<u8>::new());
+        record.write(&mut cursor).expect("failed to write record");
+        let written_len = cursor.position();
+        cursor.set_position(0);
+        let mut reader = TfRecordReader::new(cursor);
+        let output_record = reader.read_record().expect("read_record");
```

Comment:
nit: `...expect("read_record");` --> `expect("failed to read record");` ?

Reply:
I’m not certain what the prevailing style is here, but it renders like
this:

```
thread '<test_name>' panicked at 'read_record: BadLengthCrc(Truncated)', tensorboard/data/server/tf_record.rs:123:45
```

so I think it’s already clear enough. The fact that it’s panicking and
the test fails means that it’s failed.

If this were in non-test code and I were using `expect` for a condition
that _should_ never happen (i.e., an assertion), I’d be more careful to
provide clear information and context. But for a test that’s completely
self-contained, I think that this is fine.